websock/session.nim

```diff
@@ -433,6 +430,39 @@ proc recv*(
 
   return res
 
+proc recvTextMsg*(
+  ws: WSSession,
+  size = WSMaxMessageSize): Future[string] {.async.} =
+  let msg = await ws.recvMsg(size)
+
+  if ws.binary:
+    raise newException(WSInvalidOpcodeError, "Got binary when text expected!")
+
+  if validateUTF8(msg.toOpenArray(0, msg.high)) == false:
```

Comment:
Moving `validateUTF8` to here does not really fix #85, because the `size` parameter can easily trigger the bug if it's value lies in the middle of valid UTF-8 sequence. `validateUTF8` need to be modified into stateful validator. but I guess that can be fixed in next PR.

Reply:
It doesn't, `size` is the maximum size of the message, if the total amount of bytes read exceeds size it will throw either way. In other words, you can't have partial messages which is what's all of this is about.